AUFN OpenStack Antelope

oneswig opened the pull request:
Some fixes and improvements for the AUFN workshop lab deployment and configuration

oneswig committed:
Add security groups for AUFN services

oneswig committed:
Bootstrap for CHPC lab

oneswig committed:
Fixes for deploying Antelope (2023.1)

Ubuntu by default
Add initial TENKS provision to a-seed-from-nothing
Migrate to quay.io and new tag naming scheme
Fix for loss of DHCP on Ubuntu/Netplan instances
Give lab-specific names to security group and ssh key
Disable interactive dialog for service restart on Jammy package update

MoteHue commented on a-seed-from-nothing.sh:
```diff
@@ -145,6 +161,11 @@ if ! ./dev/seed-deploy.sh; then
     ./dev/seed-deploy.sh
 fi
 
+# Run TENKS
+cd ~/kayobe
+export TENKS_CONFIG_PATH=config/src/kayobe-config/tenks.yml
+./dev/tenks-deploy-overcloud.sh ./tenks
+
```
Do we always want to do this now, or was this just to speed up this weeks' sessions? I can see the benefit of having these deployed in advance, especially given we later have users use tenks to deploy a new hypervisor

oneswig replied:
I think so, it's a liability...

MoteHue commented on a-seed-from-nothing.sh:
```diff
@@ -4,7 +4,7 @@
 
 # Cloud User: cloud-user (CentOS) or ubuntu?
-CLOUD_USER=cloud-user
+CLOUD_USER=ubuntu
```
Are you sure we want to change the default to ubuntu?

Alex-Welsh replied:
Why not just remove the default user https://github.com/stackhpc/terraform-aufn/pull/16

Alex-Welsh committed:
Automate default user selection and OS evaluation

Alex-Welsh committed:
Update README

Alex-Welsh commented:
Can we just make this the main branch, rather than merging it into Yoga?

markgoddard committed:
Merge pull request #17 from stackhpc/update-readme

Update README

jackhodgkiss committed:
feat: optionally create A records for labs

Lab instances may have their fixed IPv4 address used for DNS A record
within Designate. Minor change that improves accessing and managing lab
instances.

jackhodgkiss committed:
Update vars.tf

Co-authored-by: Mark Goddard <mark@stackhpc.com>

jackhodgkiss committed:
feat: remove `depends_on` from `lab_zone`

markgoddard committed:
Merge pull request #18 from stackhpc/dns-for-fixed-ipv4

feat: optionally create A records for labs

markgoddard committed:
Merge pull request #16 from stackhpc/automate-os

Automate default user selection and OS evaluation

Alex-Welsh committed:
Refactor to use Beokay

Beokay is a tool for creating production kayobe environments. It
provides a clear source tree with a simple python script.

This change primarily moves the source code and python virtual
environments into a new ~/deployment directory, and separates each
source code directory (Kayobe, Kayobe-Config, Kolla-Ansible, Tenks).

Alex-Welsh committed:
Merge pull request #19 from stackhpc/beokay-refactor

Beokay Refactor
